src/main/java/InfrastructureManager/MatchMaking/MatchMaker.java

```diff
@@ -119,16 +616,52 @@ public String read() throws Exception {
         return clientList;
     }
 
-    public Map<EdgeClient, EdgeNode> getMapping() {
+    public Map<String, String> getMapping() {
         return mapping;
     }
 
+    /*
+    ----------------------Just making sure input client are in client list---------------
+     */
     private EdgeClient getClientByID(String clientID) throws Exception {
-        for (EdgeClient client : this.clientList) {
-            if (client.getId().equals(clientID)) {
-                return client;
+        for (EdgeClient thisClient : this.clientList) {
+            if (thisClient.getId().equals(clientID)) {
+                return thisClient;
             }
         }
-        throw new Exception("No client found");
+        throw new Exception("can not find this [clientID] in the list");
     }
+
+    private EdgeNode getNodeByID(String nodeID) throws Exception {
+        for (EdgeNode thisNode : this.nodeList) {
+            if (thisNode.getId().equals(nodeID)) {
+                return thisNode;
+            }
+        }
+        throw new Exception("can not find this [nodeID] in the list");
+    }
+
+    /**
+     * get a list of client that connected to nodeID
+     *
+     * @param nodeID
+     * @return list of client assigned to nodeID
+     * @author Zero
+     */
+    private ArrayList<String> getAssignedClient(String nodeID) {
```

Comment:
I believe you are returning a list of clients. Change name to getAssignedClientList or just getAssignedClients.

Reply:
done in 09ce855